data-upgrade-users/src/main/java/org/exoplatform/migration/UsersLastLoginTimeMigration.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2022 eXo Platform SAS.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.exoplatform.migration;
+
+import org.exoplatform.commons.search.index.IndexingService;
+import org.exoplatform.commons.upgrade.UpgradeProductPlugin;
+import org.exoplatform.commons.utils.CommonsUtils;
+import org.exoplatform.commons.utils.ListAccess;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.organization.OrganizationService;
+import org.exoplatform.services.organization.User;
+import org.exoplatform.social.core.identity.model.Identity;
+import org.exoplatform.social.core.identity.model.Profile;
+import org.exoplatform.social.core.identity.provider.OrganizationIdentityProvider;
+import org.exoplatform.social.core.jpa.search.ProfileIndexingServiceConnector;
+import org.exoplatform.social.core.manager.IdentityManager;
+import org.exoplatform.social.core.profile.ProfileFilter;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class UsersLastLoginTimeMigration extends UpgradeProductPlugin {
+  private static final Log    LOG = ExoLogger.getExoLogger(UsersLastLoginTimeMigration.class);
+
+  private OrganizationService organizationService;
+
+  private IdentityManager     identityManager;
+
+  public UsersLastLoginTimeMigration(OrganizationService organizationService,
+                                     IdentityManager identityManager,
+                                     InitParams initParams) {
+    super(initParams);
+    this.organizationService = organizationService;
+    this.identityManager = identityManager;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    LOG.info("Start upgrade process to add lastLoginTime in user profile");
+    long startupTime = System.currentTimeMillis();
+    int limit = 3;
```

Comment:
your limit should be different than 3. 
```suggestion
    int limit = 100;
```